App/Containers/Contact.tsx

```diff
@@ -67,29 +80,61 @@ class ContactModal extends React.Component<Props> {
           >
             {this.props.displayName}
           </Text>
+          <Button
+            text={buttonText}
+            disabled={buttonDisabled}
+            onPress={this.props.isContact ? this.onRemove : this.onAdd}
+          />
         </View>
         <ScrollView style={styles.threadsList}>
           <Text style={styles.threadsTitle}>
             {this.props.threadThumbs.length > 0 ? 'Sharing in Groups:' : 'Not part of any shared groups'}
           </Text>
           {this.props.threadThumbs.map((thread, i) => (
-            <TouchableOpacity key={i} onPress={this.navigate(thread.id)}>
+            <TouchableOpacity key={i} onPress={this.navigateToThread(thread.id)}>
             <PhotoWithTextBox key={i} text={thread.name} photo={thread.thumb} />
             </TouchableOpacity>
           ))}
         </ScrollView>
       </View>
     )
   }
+
+  onRemove = () => {
+    this.props.removeContact(this.props.navigation.getParam('contact').address)
+  }
+
+  onAdd = () => {
+    this.props.addContact(this.props.navigation.getParam('contact'))
+  }
```

Comment:
And same in the calls inside of these

Reply:
Oof idk how I missed this. Fixing it right now.